webapp/src/app/admin/devices/devices.component.ts

```diff
@@ -153,23 +200,43 @@ export class DevicesComponent implements OnInit {
     templateUrl: "device-add.html",
 })
 export class DeviceAddComponent {
+    typeStr: string;
+
     constructor(
         public dialogRef: MatDialogRef<DeviceAddComponent>,
         @Inject(MAT_DIALOG_DATA) public data: string) {
     }
 }
 
 @Component({
-    selector: "device-show",
-    templateUrl: "device-show.html",
+    selector: "show",
+    templateUrl: "show.html",
 })
-export class DeviceShowComponent {
+export class ShowComponent {
```

Comment:
Nice! I find it more readable this way (keeping related items together, short names, ...)
Now, to be picky, why then the `rename` to `renameDevice` transformation (amongs other)?

Reply:
Good catch. `rename` handles both devices and recovery, so it should not be `renameDevice`. The other methods only handle one case, so I keep them like that.